drivers/nutdrv_qx.c

```diff
@@ -2017,7 +2132,7 @@ void	upsdrv_initups(void)
 		dstate_setinfo("ups.productid", "%04x", usbdevice.ProductID);
 
 		/* Check for language ID workaround (#2) */
-		if (langid_fix != -1) {
+		if ((langid_fix != -1) && (!getval("noscanlangid"))) {
```

Comment:
You add AND comparison operator which may cause error for other protocols. Are you sure about that?

Reply:
Seems to be okay: "if the user did not specially configure `noscanlangid` option, then..." - for majority of cases this is going to be true, unless users specially did something.